lib/src/main/kotlin/ISolver.kt

```diff
@@ -0,0 +1,9 @@
+interface ISolver<TAction> {
```

Comment:
@JunTaoLuo quick google search tells me that we should not use C# conventions in JVM, that is interfaces shouldn't be named `IXyz` perhaps just `xyz` or `xyzInterface` [see here](https://stackoverflow.com/questions/541912/interface-naming-in-java)

Reply:
Sounds good.